### PR TITLE
Use Net-http-persistent for persistent connection

### PR DESCRIPTION
**Pull request by bhelx**
Should discuss some configuration and dependency management here.

Tested against this script:

```ruby
require 'recurly'
require 'benchmark'

API_KEY = 'KEY'
SITE_ID = 'subdomain-SUBDOMAIN'

@client = Recurly::Client.new(api_key: API_KEY, site_id: SITE_ID)

# Get first 30 ids
ids = @client.list_accounts(limit: 30).each_page.first.map(&:id)

measurement = Benchmark.measure do
  # Fetch 30 accounts
  # Obviously we could use `ids=` in list_accounts but this
  # is unoptimized for benchmarking purposes
  ids.each do |id|
    acct = @client.get_account(account_id: id)
  end
end
puts measurement
```

Without persistent:
`0.520000   0.040000   0.560000 ( 11.453849)`
With persistent:
`  0.390000   0.030000   0.420000 (  4.655093)`

With persistent, the wall-clock time is over halved.

### CODE REVIEW

**Comment by drewish** [comment]
Yeah https://github.com/drbrain/net-http-persistent/issues/80 makes me wonder if we want to pull in that dependency.

**Comment by bhelx** [comment]
@drewish yeah it's odd. Not sure why the project doesn't have release notes. But perhaps we can use `2.9`? Also, it might make sense to support this as an optional adapter.

**Comment by bhelx** [comment]
Notes: https://github.com/drbrain/net-http-persistent/blob/master/History.txt

**Comment by bhelx** [comment]
@cbarton right, thinking that might be the best option. Maybe also let them provide the version of `net-http-persistent` they use as well.

**Comment by cbarton** [comment]
Or just a symbol and we provide the adapter integration

File: lib/recurly/client/adapter.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          end
+          http.open_timeout = 50
+          http.read_timeout = 60
+        end
+      end
+    end
+
+    module NetHttpAdapter
+      protected
+      def configure_adapter(http)
```
Comment (drewish) [inline lib/recurly/client/adapter.rb]: realizing this probably should have been `faraday`

Comment (bhelx) [reply lib/recurly/client/adapter.rb]: https://github.com/recurly/recurly-client-ruby/commit/bb46c8869defe88df593d4e890d5a9de73713550

**Pull request merged**